academic-services/knowledge-exploration-service/tutorial-grammar-design.md

```diff
@@ -0,0 +1,401 @@
+---
+title: Build a library search application
+description: Step by step tutorial to design MAKES grammar for custom data
+ms.topic: tutorial
+ms.date: 11/16/2020
+---
+
+# Build a library search
+
+This tutorial illustrates how to build a library search using the linked publication records from [link private publication records with MAKES entities tutorial](tutorial-entity-linking.md) and concepts from [library browser with contextual filters tutorial](tutorial-entity-linking.md). You will learn how to:
+
+- Design a MAKES schema tailored for publication search
+- Design a MAKES grammar to process search queries
+- Build, compile, and deploy a MAKES instance with custom index and grammar
+- Create a frontend client for publication search using MAKES APIs.
+
+![Library search application](media/privateLibraryExampleApp-search-homepage.png)
+
+## Prerequisites
+
+- [Microsoft Academic Knowledge Exploration Service (MAKES) subscription](get-started-setup-provisioning.md) (release version after 2020-11-23)
+- Completion of [link private publication records with MAKES entities tutorial](tutorial-entity-linking.md)
+- Completion of [build a library browser with contextual filters tutorial](tutorial-schema-design.md)
+- Read through the [define index schema how-to guide](how-to-index-schema.md)
+- Read through the [define grammar how-to guide](how-to-grammar.md)
+- Download the [sample search schema for linked private library publications](samplePrivateLibraryData.linked.search.schema.json)
+- Download the [sample search grammar for linked private library publications](samplePrivateLibraryData.linked.search.grammar.xml)
+
+## Design a schema for search
+
+When designing a new schema for search, it's important to evaluate the following:
+
+- What entity attributes can appear in search queries?
+- How will the entity attributes appear in search queries?
+
+### Publication search schema design goal
+
+To determine what attributes that need to be indexed, we need to drill into the types of search queries we will process. We can map out what search queries need to be by supported by going through the search scenarios. For this tutorial, we want to enable users to
+
+- Search by categorical attributes, such as fields of study, author and affiliation.
+- Search by keywords extracted from abstact and title
+- Search by identifiable attributes, such as the publication's title, DOI, and the full text url from the original records.
```

Comment:
These are semantic attributes, same as FOS, authors, affiliation, etc. that you outlined in the first bullet point. Calling them out separately here is confusing and unnecessary